File: src/main/java/pl/tsimur/jpatest/service/SubscriptionService.java
package pl.tsimur.jpatest.service;

import pl.tsimur.jpatest.model.Subscription;
import pl.tsimur.jpatest.model.Tariff;

public interface SubscriptionService {

    Subscription find(String name);

    Subscription findTitle(String title);

    Subscription find(Float price);

    void updateTitle(Tariff name, String title);

    void updatePrice(String title, Float price);
}

File: src/main/kotlin/pl/timur/jpatest/service/SubscriptionService.kt
package pl.timur.jpatest.service

import pl.timur.jpatest.model.Subscription
import pl.timur.jpatest.model.Tariff

interface SubscriptionService {
    fun find(name: String): Subscription?
    fun findTitle(title: String): Subscription?
    fun find(price: Float): Subscription?
    fun updateTitle(name: Tariff, title: String)
    fun updatePrice(title: String, price: Float)
}